wordpress-mobile/WordPress-iOS
# Allow new password input keyboard to be dismissed

Addresses https://github.com/wordpress-mobile/WordPress-iOS/issues/17927

### To test

1. Navigate to the welcome screen
2. Tap the button to log-in/sign-up
3. Enter a new email address 
4. Close the WordPress app and open your email client on the device
4. Tap the link in the sign-up email and expect the app to open again
5. On the screen that allows the user to set their Display Name, Username and Password:
  a. Tap the Password field
  b. Notice the keyboard "done" button is enabled and closed the keyboard when tapped (before this PR, the button was disabled)
  c. Interact with the other fields and verify they look and behave correctly

## Regression Notes
> 1. Potential unintended areas of impact

This change should only affect the Password field, but since it shares logic with the Display Name and Username fields, we should check for regressions there. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)

[`SignupTests`](https://github.com/wordpress-mobile/WordPress-iOS/blob/57adae53b8e2f3dfaf9320af887a52b944beba19/WordPress/WordPressUITests/Tests/SignupTests.swift#L19-L30) will help catch regressions related to setting a password.

3. What automated tests I added (or what prevented me from doing so)

Writing an automated test here would be very challenging because the change involves the virtual keyboard which AFAIK is hard to control in tests.

PR submission checklist:
